src/ResourceManagement/Sql/SqlEncryptionProtectorImpl.cs

```diff
@@ -0,0 +1,221 @@
+        private ISqlManager sqlServerManager;
+        private string resourceGroupName;
+        private string sqlServerName;
+        private string serverKeyName;
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="parent">Reference to the parent of this external child resource.</param>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses firewall rule operations.</param>
+        ///GENMHASH:037B6E72FFBB7224CDF82D509814DF07:925E12E2AB778ADEA6FAE0AE9F9184C7
+        internal SqlEncryptionProtectorImpl(SqlServerImpl parent, EncryptionProtectorInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, parent)
+        {
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = parent.ResourceGroupName;
+            this.sqlServerName = parent.Name;
+            this.serverKeyName = innerObject?.Name;
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name.</param>
+        /// <param name="sqlServerName">The parent SQL server name.</param>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses firewall rule operations.</param>
+        ///GENMHASH:B9BEF73056A0E29FFD8BF2E062D7BC15:43162FE085CFEAFDA76E5D4BC4A00355
+        internal SqlEncryptionProtectorImpl(string resourceGroupName, string sqlServerName, EncryptionProtectorInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = resourceGroupName;
+            this.sqlServerName = sqlServerName;
+            this.serverKeyName = innerObject?.Name;
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses firewall rule operations.</param>
+        ///GENMHASH:4DD6445361F09E0B185DE0E8D24170D3:81D0EDB620000F6B86DC1403C1548E6A
+        internal SqlEncryptionProtectorImpl(EncryptionProtectorInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.sqlServerManager = sqlServerManager;
+            if (innerObject != null && innerObject.Id != null)
+            {
+                if (innerObject.Id != null)
+                {
```

Comment:
you already check for this not being null 2 lines above, right?

Reply:
Fixed